### ISSUE TITLE
Host app fails to serve remote apps for dynamic module federation in angular

### ISSUE THREAD

**Issue by kenLovesToCode** [nrwl/nx#19893]
### Current Behavior

Running `yarn nx serve shell --devRemotes=app1,app2` in the terminal produces this output below
> nx run shell:serve:development --devRemotes=app1,app2
>  NX  Starting module federation dev-server for shell with 0 remotes

### Expected Behavior

In the past version of nx@16.10.0, this will run the host(shell) together with 2 remotes(app1, app2)

### GitHub Repo

https://github.com/kenLovesToCode/test-for-bug-nx-angular-monorepo-dynamic-federation

### Steps to Reproduce

1. Run `npx create-nx-workspace sample --packageManager=yarn`
Need to install the following packages:
  create-nx-workspace@17.0.2
Ok to proceed? (y) y

 >  NX   Let's create a new workspace [https://nx.dev/getting-started/intro]

√ Which stack do you want to use? · angular
√ Integrated monorepo, or standalone project? · integrated
√ Application name · tester
√ Default stylesheet format · scss
√ Test runner to use for end to end (E2E) tests · cypress
√ Would you like to use Standalone Components in your application? · No
√ Would you like to add routing? · Yes
√ Enable distributed caching to make your CI faster · Yes

 >  NX   Creating your v17.0.2 workspace.
 
 2. Run `yarn nx g @nx/angular:host shell --remotes=app1,app2 --dynamic`
√ Which stylesheet format would you like to use? · scss
√ Which E2E test runner would you like to use? · cypress
√ What should be the project name and where should it be generated? · shell @ apps/shell

### Nx Report

```shell
Node   : 18.14.1
   OS     : win32-x64
   yarn   : 1.22.19

   nx                 : 17.0.2
   @nx/js             : 17.0.2
   @nx/jest           : 17.0.2
   @nx/linter         : 17.0.2
   @nx/eslint         : 17.0.2
   @nx/workspace      : 17.0.2
   @nx/angular        : 17.0.2
   @nx/cypress        : 17.0.2
   @nx/devkit         : 17.0.2
   @nx/eslint-plugin  : 17.0.2
   @nrwl/tao          : 17.0.2
   @nx/web            : 17.0.2
   @nx/webpack        : 17.0.2
   typescript         : 5.1.6
```

### Operating System

- [ ] macOS
- [ ] Linux
- [X] Windows
- [ ] Other (Please specify)

**Comment by RichFinn-WTW** [comment]
Just came to open this same issue - thanks @kenLovesToCode. 

Workarounds I've found for now:

1. define the remotes needed in the host's module-federation.config.js:

    ```js 
    module.exports = {
      name: 'host-application',
      remotes: ['remote-app-1'],
    };
    ```

    this starts up the remotes defined in the property and produces the desired result

    >  NX  Starting module federation dev-server for host-application with 1 remotes
      

1. open a new terminal and run `nx serve [remote-project]` for each remote needed

**Comment by kenLovesToCode** [comment]
@RichFinn-WTW So regardless of being a dynamic module federation, it requires to have the remote on the remotes list? Then this docs need to be updated [Advanced Micro Frontends with Angular using Dynamic Federation](https://nx.dev/recipes/angular/dynamic-module-federation-with-angular#change-how-remotes-are-loaded ). 

Thanks for this solution, it works on development environment. I'll try this on different environments.

**Comment by RichFinn-WTW** [comment]
@kenLovesToCode, I think this is still a bug. I'm guessing the `devRemotes` property _should_ be working without the need for the remotes to be listed in the module-federation.config.js file. My suggestion is just a workaround until it's fixed.

**Comment by RichFinn-WTW** [comment]
can confirm that this issue is resolved in nx 17.0.3 with pr #19894, resolving #19880, which this issue duplicates
